server/src/main/java/org/elasticsearch/common/geo/GeometryTreeReader.java

```diff
@@ -33,28 +34,40 @@
  */
 public class GeometryTreeReader {
 
-    private final BytesRef bytesRef;
+    private final ByteBufferStreamInput input;
 
     public GeometryTreeReader(BytesRef bytesRef) {
-        this.bytesRef = bytesRef;
+        this.input = new ByteBufferStreamInput(ByteBuffer.wrap(bytesRef.bytes, bytesRef.offset, bytesRef.length));
     }
 
-    public boolean containedInOrCrosses(int minLon, int minLat, int maxLon, int maxLat) throws IOException {
-        ByteBufferStreamInput input = new ByteBufferStreamInput(
-            ByteBuffer.wrap(bytesRef.bytes, bytesRef.offset, bytesRef.length));
+    public Extent getExtent() throws IOException {
```

Comment:
This method is no longer needed, right?

Reply:
so, this method is to be used by aggregations like `geo_bounds` in the future, so I figured I would throw it in here while I'm at it. happy to remove and only add when used

Reply:
In this case we should probably add a test for it and replace reading individual components with invoking `Extent(StreamInput).

Reply:
++ will follow-up